src/Shared/Microsoft.DotNet.Scaffolding.Shared/T4Templating/ITextTransformation.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.CodeDom.Compiler;
+using System.Collections.Generic;
+
+namespace Microsoft.DotNet.Scaffolding.Shared.T4Templating
+{
+    /// <summary>
+    ///     This is currently an internal API that supports scaffolding. Use with caution.
+    /// </summary>
+    public interface ITextTransformation
+    {
+        /// <summary>
+        /// Session object that can be used to transmit information into a template.
+        /// </summary>
+        IDictionary<string, object> Session { get; set; }
```

Comment:
>Session [](http://example.com/codeflow?start=36&length=7)

does this object need this Session here? Or InvokeTemplate.InvokeTemplate should receive this dictionary of params?

Reply:
The ITextTransformation needs to hold the `Session`. It's hard to see why currently but the generated T4 templates use it get parameters.